commands/init.go

```diff
@@ -39,32 +39,38 @@ func transformInitArgs(args *Args) error {
 		return nil
 	}
 
-	var (
-		name   string
-		newDir bool
-		err    error
-	)
+	var err error
+	dirToInit := "."
+	hasValueRegxp := regexp.MustCompile("^--(template|separate-git-dir|shared)$")
 
-	if args.IsParamsEmpty() {
-		name, err = utils.DirName()
-		if err != nil {
-			return err
+	// Find the first argument that isn't related to any of the init flags.
+	// We assume this is the optional `directory` argument to git init.
+	for i := 0; i < args.ParamsSize(); i++ {
+		arg := args.Params[i]
+		if hasValueRegxp.MatchString(arg) {
+			i++
+		} else if !strings.HasPrefix(arg, "-") {
+			dirToInit = arg
+			break
 		}
-	} else {
-		name = args.LastParam()
-		newDir = true
 	}
 
-	project := github.NewProject("", name, "")
+	dirToInit, err = filepath.Abs(dirToInit)
```

Comment:
In general, I like the idea of reducing special cases with a default value "." for `dirToInit`.